app/interface/deps/client_deps.py

```
@@ -0,0 +1,31 @@
+from fastapi import Body, HTTPException
+
+from app.interface.schemas.common.requests import LoginRequest
+from app.interface.clients import KutisClient, LmsClient, BarunClient
+from app.infrastructure.exceptions import ClientException
+
+
+def client_provider_factory(
+    client_cls: type[Union[KutisClient, LmsClient, BarunClient]],
+) -> Callable[[], AsyncGenerator[Union[KutisClient, LmsClient, BarunClient], None]]:
+    async def get_client(
+        request: LoginRequest = Body(...),
```

Ruff (0.12.2)

Do not perform function call `Body` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)